### PR TITLE
Completed api-registration exercise - Noah Lett

### PR DESCRIPTION
**Pull request by NoahLett**
[Untitled_ Dec 20, 2022 8_12 PM.webm](https://user-images.githubusercontent.com/110089434/208819921-cf186e04-e5b7-41d5-b660-b38daf9dd06b.webm)

### CODE REVIEW

File: api-registration/server/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+  argon2.hash(password)
+    .then(hashedPassword => {
```
Comment (Zhiwutian) [inline api-registration/server/index.js:27]: you need to create a catch statement for this .then, the one nested in the query chain will not catch an error here

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
Tiny update!

File: api-registration/server/index.js (hunk 1 above)
Comment (NoahLett) [reply api-registration/server/index.js:27]: Just added another catch statement!

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**